src/ui/layer_list_panel.ts

```diff
@@ -103,6 +106,66 @@ export class LayerVisibilityWidget extends RefCounted {
   }
 }
 
+class LayerColorWidget extends RefCounted {
+  element = document.createElement("div");
+  elementWrapper = document.createElement("div");
+
+  constructor(
+    public panel: LayerListPanel,
+    public layer: ManagedUserLayer,
+  ) {
+    super();
+    const { element, elementWrapper } = this;
+    element.className = "neuroglancer-layer-list-panel-color-value";
+    elementWrapper.className =
+      "neuroglancer-layer-list-panel-color-value-wrapper";
+    elementWrapper.appendChild(element);
+
+    this.registerDisposer(
+      observeWatchable((layerColorEnabled) => {
+        elementWrapper.style.display = layerColorEnabled ? "block" : "none";
+        if (this.layer.supportsLayerBarColorSyncOption) {
+          elementWrapper.classList.remove("unsupported");
+          element.classList.remove("unsupported");
+        } else {
+          elementWrapper.classList.add("unsupported");
+          element.classList.add("unsupported");
+        }
+      }, panel.sidePanelManager.viewerState.enableLayerColorWidget),
+    );
+    if (!this.layer.supportsLayerBarColorSyncOption) {
```

Comment:
I think ideally this needs to be watched for changes, because layer types can change. You can end up with something like this currently:

![image](https://github.com/user-attachments/assets/ee84b1dd-0715-49fc-ad53-b1b492c0fff8)

Where layer 12 changed type and doesn't match in the two bars

Reply:
I think that would also incorporate the logic in for lines 127 - 132, and remove it from there. The object could track what style the color widget should be when displayed. And then watch for the global setting for whether to show the widget or not. As opposed to checking the style when the global setting changes.